src/sentry/utils/pytest/fixtures/bad_model_signal_handlers.py

```diff
@@ -0,0 +1,96 @@
+import pytest
+
+
+def _format_msg(bad_signal_handlers):
+    msg = """\
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+We have detected that you are spawning a celery task inside of a model signal
+handler such as post_save or post_delete without using transaction.on_commit:
+"""
+
+    for signal_sender, task_self in bad_signal_handlers:
+        msg += f"  - A change to model {signal_sender} spawning a task {task_self.__name__}\n"
+
+    msg += """
+We saw that this kind of signal handler is often subtly broken in how it
+interacts with database transactions. Typically the fix is to change the signal
+handler to use django.db.transaction.on_commit:
+
+    def my_signal_handler(...):
+        # Before
+        handle_model_changes.apply_async(...)
+        # After
+        transaction.on_commit(lambda: handle_model_changes.apply_async(...))
+
+For an example of a real-world fix, see
+https://github.com/getsentry/sentry/pull/35523, and for the PR that introduced
+this fixture see https://github.com/getsentry/sentry/pull/35671
+
+If you think this doesn't apply to this test, use the
+`bad_model_signal_handlers` fixture like so at the end of the test:
+
+    class MyTestCase(TestCase):
+        def test_very_special(self):
+            self.bad_model_signal_handlers.clear()
+
+Or like this in pytest-based tests:
+
+    def test_very_special(bad_model_signal_handlers):
+        # Do something odd with models here
+
+        bad_model_signal_handlers.clear()
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"""
+
+    return msg
+
+
+@pytest.fixture(autouse=True)
+def bad_model_signal_handlers(monkeypatch):
+    from threading import local
+
+    from celery.app.task import Task
+    from django.db import transaction
+    from django.db.models.signals import ModelSignal
```

Comment:
these are all 3rd party imports and all for an autouse fixutre.  why not import them at the top of the module?

Reply:
you are right